frontend/src/pages/projects/screens/spawner/imageSelector/ImageVersionSelector.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { FormGroup, Select, SelectOption, Text, Title } from '@patternfly/react-core';
+import { ImageVersionSelectDataType, ImageVersionSelectOptionObjectType } from '../types';
+import {
+  checkTagBuildValid,
+  compareImageVersionOptionOrder,
+  getAvailableVersionsForImageStream,
+  getImageVersionDependencies,
+  getImageVersionSoftwareString,
+  isImageVersionSelectOptionObject,
+} from '../spawnerUtils';
+import ImageVersionTooltip from './ImageVersionTooltip';
+import { getDashboardMainContainer } from '../../../../../utilities/utils';
+
+type ImageVersionSelectorProps = {
+  data: ImageVersionSelectDataType;
+  selectedImageVersion?: ImageVersionSelectOptionObjectType;
+  onImageVersionSelect: (selection: ImageVersionSelectOptionObjectType) => void;
+};
+
+const ImageVersionSelector: React.FC<ImageVersionSelectorProps> = ({
+  selectedImageVersion,
+  onImageVersionSelect,
+  data,
+}) => {
+  const [versionSelectionOpen, setVersionSelectionOpen] = React.useState<boolean>(false);
+
+  const { imageStream, versionOptions, buildStatuses } = data;
+
+  if (!imageStream || getAvailableVersionsForImageStream(imageStream, buildStatuses).length <= 1) {
```

Comment:
It's interesting you are checking the versions through a function here to see how many they are, but then don't use it in the rest of the logic? 🤔 Are `versionOptions` not enough?

Reply:
For posterity, Juntao and I talked -- `versionOptions` are the "all options" and `getAvailableVersionForImageStream` is to check if we have options to be enabled / selectable. If you have 5 options, and only 1 is selectable -- we don't need a dropdown. 

@DaoDaoNoCode my recommendation would be to see how -- when you restructure the data -- we make versions something we can check availability on -- still a little weird for this function to be used only for an if-statement -- but if you can't cleanly see a path forward we can talk or you can keep this code (I understand its current usefulness).